runtime-modules/content/src/lib.rs

```diff
@@ -674,10 +674,24 @@ decl_module! {
 
             // upload to storage
             if let Some(upload_assets) = params.assets.as_ref() {
+                let dyn_bag = DynamicBagIdType::<T::MemberId, T::ChannelId>::Channel(channel_id);
+                let bag_id = BagIdType::from(dyn_bag.clone());
+
+                let upload_params = UploadParametersRecord {
+                    bag_id: bag_id.clone(),
+                    object_creation_list: upload_assets.object_creation_list.clone(),
+                    deletion_prize_source_account_id: sender.clone(),
+                    expected_data_size_fee: upload_assets.expected_data_size_fee,
+                };
+
+                // verify that bag exists before hand
+                Storage::<T>::can_create_dynamic_bag_with_objects_constraints(&dyn_bag, &None, &upload_params)?;
```

Comment:
1. The comment here seems invalid, the function does not verify whether the bag exists, but whether it can be created
2. I'm not very familiar with the runtime development, but wouldn't that be considerably computionaly wasteful to run this function twice? (first here and then inside `create_dynamic_bag_with_objects_constraints`) @shamil-gadelshin also pointed out that the result of the random bucket selection may be different when its ran the second time.